weave_query/weave_query/weave_http.py

```diff
@@ -128,6 +129,29 @@ async def download_file(
                         r.status, "Download failed"
                     )
 
+    async def query_traces(
+        self,
+        url: str,
+        payload: typing.Optional[dict],
+        headers: typing.Optional[dict[str, str]] = None,
+        cookies: typing.Optional[dict[str, str]] = None,
+        auth: typing.Optional[aiohttp.BasicAuth] = None,
+    ) -> list[dict]:
+        with tracer.trace("query_traces"):
+            results = []
+            async with self.session.post(url, json=payload, headers=headers, cookies=cookies, auth=auth) as response:
+                if response.status == 200:
+                    async for line in response.content:
+                        if line:
+                            decoded_line = line.decode('utf-8').strip()
```

Comment:
subtly here: not all trace server endpoints return jsonl. therefore, the pass in of URL feels a little odd here. It is possible for a future dev to pass in another url for a different endpoint that would result in breakage.